include/stdexec/execution.hpp

```diff
@@ -243,8 +294,35 @@ namespace stdexec {
           noexcept(nothrow_tag_invocable<get_env_t, const _EnvProvider&>)
           -> tag_invoke_result_t<get_env_t, const _EnvProvider&> {
           static_assert(queryable<tag_invoke_result_t<get_env_t, const _EnvProvider&> >);
+          __check_sender_version<remove_cvref_t<_EnvProvider>>();
```

Comment:
I'm trying to figure how this doesn't give sender warnings on receivers passed to `get_env`. Any hints?

Reply:
Ah, you're right, receiver types do trigger this. I'd need to add another half baked concept to sense for a completion signature operation on the type.

Reply:
I moved the deprecation logic inside of the connect CPO, invoking separate checks on each of the deduced Sender and Receiver types.